src/opm/output/eclipse/AggregateGroupData.cpp

```diff
@@ -462,8 +462,8 @@ namespace {
 void
 Opm::RestartIO::Helpers::groupMaps::
 currentGrpTreeNameSeqIndMap(const Opm::Schedule&                        sched,
-                            const size_t                                simStep,
```

Comment:
Although people have different opinions, I do think `const` here is good to protected the value is modified within the function. It also applies to a few other circumstances in this PR.

Reply:
Well the compiler guys seem to think differently. What exactly does the const prevent and what would the consequence be if it is dropped. 

The problem is that some warnings will really indicate problems and bugs. But if we do not try to eliminate as many warnings as possible, then chances are they will simply be ignored.

Reply:
> try to eliminate as many warnings as possible, 

Yes. I totally agree.  

Can you share a warning message about this?

Reply:
Thanks for asking and being so persistent. I could not find the warning and it must have been issued by my head rather than my compiler. The latter agrees with you.

I added a patch that reintroduces the `const`.